src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -80,38 +85,67 @@ private static boolean doesPrefixHaveOneValue(ArgumentMultimap argumentMultimap,
-        List<String> outputList = new ArrayList<>();
+
 
         if (PREFIX_NAME.equals(prefix) && argumentMultimap.getValue(PREFIX_NAME).isPresent()) {
-            for (String s : argumentMultimap.getAllValues(PREFIX_NAME)) {
-                String parsedNameAsString = ParserUtil.parseName(s).toString();
-                outputList.add(parsedNameAsString.trim());
-            }
+            return parseValueHelper(argumentMultimap, PREFIX_NAME);
         } else if (PREFIX_ADDRESS.equals(prefix) && argumentMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
-            for (String s : argumentMultimap.getAllValues(PREFIX_ADDRESS)) {
-                String parsedAddrAsString = ParserUtil.parseAddress(s).toString();
-                outputList.add(parsedAddrAsString.trim());
-            }
-
+            return parseValueHelper(argumentMultimap, PREFIX_ADDRESS);
         } else if (PREFIX_PHONE.equals(prefix) && argumentMultimap.getValue(PREFIX_PHONE).isPresent()) {
-            for (String s : argumentMultimap.getAllValues(PREFIX_PHONE)) {
-                String parsedPhoneAsString = ParserUtil.parseAddress(s).toString();
-                outputList.add(parsedPhoneAsString.trim());
-            }
+            return parseValueHelper(argumentMultimap, PREFIX_PHONE);
         } else if (PREFIX_TAG.equals(prefix) && argumentMultimap.getValue(PREFIX_TAG).isPresent()) {
-            for (String s : argumentMultimap.getAllValues(PREFIX_TAG)) {
-                String parsedTagAsString = ParserUtil.parseTag(s).toString();
-                outputList.add(parsedTagAsString.trim());
-            }
+            return parseValueHelper(argumentMultimap, PREFIX_TAG);
         } else if (PREFIX_PRICE.equals(prefix)
                 && argumentMultimap.getValue(PREFIX_PRICE).isPresent()
                 && doesPrefixHaveOneValue(argumentMultimap, PREFIX_PRICE)) {
+            List<String> outputList = new ArrayList<>();
+
             String parsedPriceAsString =
                     ParserUtil.parsePrice(argumentMultimap.getValue(PREFIX_PRICE).get()).toString();
             outputList.add(parsedPriceAsString);
+
+            return outputList;
+        } else if (PREFIX_ALL.equals(prefix)) {
+            return parseValueHelper(argumentMultimap, PREFIX_ALL);
         } else {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
+    }
+
+    /**
+     * Returns the value for the specified prefix
+     * {@code ArgumentMultimap}.
+     */
+    private static List<String> parseValueHelper(ArgumentMultimap argumentMultimap, Prefix prefix)
+            throws ParseException {
+        List<String> outputList = new ArrayList<>();
+
+        for (String s : argumentMultimap.getAllValues(prefix)) {
+            String argument = "";
+            switch (prefix.toString()) {
+            case "n/":
```

Comment:
maybe can change to `case PREFIX_NAME.toString()` and the others also for better understanding

Reply:
have clarified with mark, such an implementation raises issues within the compiler as `PREFIX_NAME.toString()` will not be considered as a constant